src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsAnalyzer.cs

```diff
@@ -144,7 +144,13 @@ public override void Initialize(AnalysisContext analysisContext)
                 }
 
                 // Compute code metrics.
-                var computeTask = CodeAnalysisMetricData.ComputeAsync(compilationContext.Compilation, compilationContext.CancellationToken);
+                var inheritanceExcludedTypes = compilationContext.Options.GetInheritanceExcludedTypeNamesOption(CA1501Rule, compilationContext.Compilation, compilationContext.CancellationToken);
```

Comment:
Using just `CA1501Rule` seems a bit weird. Should we instead make this option rule ID agnostic, and just accept category based or generic configuration? At least we should document it such so users do not try to qualify the option key name with the rule ID.

Reply:
We could use a syntax that is set in the additional text file instead of the `.editorconfig` (or at least maybe in addition). I don't really get the part with the rule id agnostic because that's how it's working for other rules. Besides, this option is really specific to this rule so I have some trouble with finding a generic way to handle this option.

Reply:
@mavasani what's your recommendation?